src/views/branchesView.ts

```diff
@@ -192,6 +194,11 @@ export class BranchesView extends ViewBase<BranchesViewNode, BranchesViewConfig>
 			() => this.setShowBranchPullRequest(false),
 			this,
 		);
+		commands.registerCommand(
+			this.getQualifiedCommand('copyCurrentBranchName'),
```

Comment:
This should live as a global command rather than inside this view (especially because this won't even get run until the view is visible for the first time).

I would recommend following this pattern: https://github.com/eamodio/vscode-gitlens/blob/main/src/commands/openBranchOnRemote.ts

You can delete the `preExecute` stuff. And like that command, you have to account for there being more than 1 repository open at a time (so the `getRepoPathOrActiveOrPrompt` call will get you the "best" repository.

Reply:
Thanks for the guidance on this! `getRepoPathOrActiveOrPrompt` is working much better and I can now select the repository I want when there is no active editor.